File: 20220720/모법답안/20220720_02.js/SwitchUpperLowerCase_test.js
import assert from 'assert'
import solution from './SwitchUpperLowerCase.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'Naver'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('nAVER', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'Kakao'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('kAKAO', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'CoFFee'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('cOffEE', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = ''

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'A'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('a', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'v'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('V', result)
    done();
  });
})

File: 20220720/모법답안/20220720_02.js/SwitchUpperLowerCase_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.SwitchUpperLowerCase.SwitchUpperLowerCase import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'Naver'

        # when
        result = solution(s)

        # then
        self.assertEqual('nAVER', result)

    def test_2(self):
        # given
        s = 'Kakao'

        # when
        result = solution(s)

        # then
        self.assertEqual('kAKAO', result)

    def test_3(self):
        # given
        s = 'CoFFee'

        # when
        result = solution(s)

        # then
        self.assertEqual('cOffEE', result)

    def test_4(self):
        # given
        s = ''

        # when
        result = solution(s)

        # then
        self.assertEqual('', result)

    def test_5(self):
        # given
        s = 'A'

        # when
        result = solution(s)

        # then
        self.assertEqual('a', result)

    def test_6(self):
        # given
        s = 'v'

        # when
        result = solution(s)

        # then
        self.assertEqual('V', result)


if __name__ == '__main__':
    unittest.main()
